### ISSUE TITLE
BUG: shap.plots.image no longer displays labels for rows other than the first one

### ISSUE THREAD

**Issue by julesvanrie** [shap/shap#4112]
### Issue Description

PR #3636 broke the display of labels on the second and subsequent rows in case:
- labels for each row are provided,
- no labels are provided but inferred from the Explainer object

Running the [Explain ResNet50 using the Partition explainer](https://shap.readthedocs.io/en/latest/example_notebooks/image_examples/image_classification/Explain%20ResNet50%20using%20the%20Partition%20explainer.html) example now no longer displays the labels on the second and subsequent rows.

### Minimal Reproducible Example

```python
# Almost straight from the example, unless marked otherwise

import json
from tensorflow.keras.applications.resnet50 import ResNet50, preprocess_input
import shap

model = ResNet50(weights="imagenet")
X, y = shap.datasets.imagenet50() 
X = X.astype(int) # Different from example to fix that actual images not showing correctly (unrelated issue)

url = "https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json"
with open(shap.datasets.cache(url)) as file:
    class_names = [v[1] for v in json.load(file).values()]

def f(x):
    tmp = x.copy()
    preprocess_input(tmp)
    return model(tmp)

masker = shap.maskers.Image("inpaint_telea", X[0].shape)
explainer = shap.Explainer(f, masker, output_names=class_names)

# Reduced number of evals for speed
# Reduced number of outputs shown to 3 (to have 2x3 instead of 2x4 so that one dimension is not a multiple of the other dimension)
shap_values = explainer(X[1:3], max_evals=20, batch_size=50, outputs=shap.Explanation.argsort.flip[:3])

# No labels on second row when running this:
shap.image_plot(shap_values)

# Same result when labels are provided
shap.image_plot(shap_values, labels=['a','b','c','d','e','f'])
shap.image_plot(shap_values, labels=[['a','b','c'],['d','e','f']])
```

### Traceback

```shell

```

### Expected Behavior

Labels should be displayed on the second row (and subsequent rows if more examples).

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [x] I'd be interested in making a PR to fix this bug

### Installed Versions

```
shap==0.48.0
shap-0.48.1.dev2 (master)
```
